sql/src/main/java/org/vertexium/sql/SqlGraphConfiguration.java

```diff
@@ -0,0 +1,137 @@
+package org.vertexium.sql;
+
+import com.google.common.base.Preconditions;
+import org.apache.commons.dbcp2.BasicDataSource;
+import org.skife.jdbi.v2.DBI;
+import org.skife.jdbi.v2.Handle;
+import org.vertexium.JavaVertexiumSerializer;
+import org.vertexium.VertexiumException;
+import org.vertexium.VertexiumSerializer;
+import org.vertexium.inmemory.InMemoryEdge;
+import org.vertexium.inmemory.InMemoryGraphConfiguration;
+import org.vertexium.inmemory.InMemoryTableElement;
+import org.vertexium.inmemory.InMemoryVertex;
+import org.vertexium.sql.collections.SqlMap;
+
+import javax.sql.DataSource;
+import java.lang.reflect.Constructor;
+import java.util.HashMap;
+import java.util.Map;
+
+public class SqlGraphConfiguration extends InMemoryGraphConfiguration {
+    protected static final String CONFIG_SERIALIZER = "serializer";
+    protected static final String CONFIG_SERIALIZER_DEFAULT = JavaVertexiumSerializer.class.getName();
+    protected static final String CONFIG_JDBC_DRIVER_CLASS = "sql.jdbc.driverClass";
+    protected static final String CONFIG_JDBC_URL = "sql.jdbc.url";
+    protected static final String CONFIG_JDBC_USERNAME = "sql.jdbc.username";
+    protected static final String CONFIG_JDBC_PASSWORD = "sql.jdbc.password";
+    protected static final String KEY_COLUMN_NAME = "id";
+    protected static final String VALUE_COLUMN_NAME = "object";
+    protected static final String VERTEX_TABLE_NAME = "vertex";
+    protected static final String EDGE_TABLE_NAME = "edge";
+    protected static final String METADATA_TABLE_NAME = "metadata";
+    protected static final String IN_VERTEX_ID_COLUMN = "in_vertex_id";
+    protected static final String OUT_VERTEX_ID_COLUMN = "out_vertex_id";
+
+    private final BasicDataSource dataSource;
+    private final SqlMap<InMemoryTableElement<InMemoryVertex>> vertexMap;
+    private final SqlMap<InMemoryTableElement<InMemoryEdge>> edgeMap;
+    private final SqlMap<Object> metadataMap;
+    private final VertexiumSerializer serializer;
+
+    public SqlGraphConfiguration(Map<String, Object> config) {
+        super(config);
+        serializer = createSerializer(config);
+        String driverClass = getConfigString(config, CONFIG_JDBC_DRIVER_CLASS);
+        String url = getConfigString(config, CONFIG_JDBC_URL);
+        String username = getConfigString(config, CONFIG_JDBC_USERNAME);
+        String password = getConfigString(config, CONFIG_JDBC_PASSWORD);
+        dataSource = new BasicDataSource();
+        dataSource.setDriverClassName(driverClass);
+        dataSource.setUrl(url);
+        dataSource.setUsername(username);
+        dataSource.setPassword(password);
+        Handle handle = new DBI(dataSource).open();
```

Comment:
I find it a little strange to open the Handle here? Should this be moved to SqlGraph?